prelude/src/core/language/Iterator.sk

```diff
@@ -205,6 +214,16 @@ mutable base class .Iterator<+T> {
     }
   }
 
+  // s.findMap(f) is Some(u) = f(t) for the first t in s where f is Some(_)
+  overridable mutable fun findMap<U>(f: T -> ?U): ?U {
+    for (t in this) {
+      u = f(t);
+      if (u.isSome()) break u
+    } else {
+      None()
+    }
+  }
```

Comment:
@mbouaziz How about this version?

Reply:
go

Reply:
Could also map().dropWhile(isNone).next(), but I guess less efficient?

Reply:
Maybe the compiler would be able to resolve and inline things enough that the efficiency would be the same, it isn't impossible, but not trivial either. But my motivation was more about the convenience of the API exposed by Iterator than pure efficiency at this point. My impression is that Iterator isn't meant to have a minimal interface, but to prefer many operations for user convenience since Iterator is in a sense a common interchange type for collections. I would be ok with aiming for minimality, but this might be the one type where I think it makes the most sense to not be minimal.